src/controllers/DeviceClaimsController.js

```diff
@@ -1,38 +1,33 @@
 // @flow
 
-import type {
-  Device,
-  DeviceRepository,
-  UserRepository,
-} from '../types';
+import type { Device, DeviceRepository } from '../types';
+import type { ClaimCodeManager } from 'spark-protocol';
 
 import Controller from './Controller';
 import httpVerb from '../decorators/httpVerb';
 import route from '../decorators/route';
 
 class DeviceClaimsController extends Controller {
   _deviceRepository: DeviceRepository;
-  _userRepository: UserRepository;
-
+  _claimCodeManager: ClaimCodeManager;
 
   constructor(
     deviceRepository: DeviceRepository,
-    userRepository: UserRepository,
+    claimCodeManager: ClaimCodeManager,
   ) {
     super();
 
     this._deviceRepository = deviceRepository;
-    this._userRepository = userRepository;
+    this._claimCodeManager = claimCodeManager;
   }
 
   @httpVerb('post')
   @route('/v1/device_claims')
   async generateClaimCode(): Promise<*> {
-    const claimCode = await this._deviceRepository.generateClaimCode(
+    const claimCode = this._claimCodeManager.addClaimCode(
```

Comment:
nit: `createClaimCode` instead.  Usually `add` means that you are setting the code as a parameter and passing it to a function.